### PR TITLE
Upgrade RethinkDB (fixes #94)

### LINKED ISSUE #94: Upgrade RethinkDB
We are on a quite-old version. 1.16.1. I'm not sure what version we could reasonably upgrade to. The most recent Python driver available through pip seems to be `2.2.0.post3`, so I'm guessing that means we don't want to upgrade the DB itself past that version? From what I've seen, I'm assuming the driver versions match the DB version, but I haven't seen that said explicitly.

Change log here: https://github.com/rethinkdb/rethinkdb/releases

Maybe @thejsj can give us some pointers on what to expect here?

### PR DESCRIPTION
**Pull request by captbaritone**
Fixes #94 

To test this, I've loaded each of the web pages and tried all the actions
I could think of. I also ran each of the tasks mentioned in `crontab` with no
problems.

Changelog here: https://github.com/rethinkdb/rethinkdb/releases?before=v1.16.1&after=v2.2.1

Order of operations to deploy
----------------------------

Upgrade the actual database package:

    apt-get install --only-upgrade rethinkdb=2.2.0

Rebuild the index:

    rethinkdb index-rebuild

Deploy this PR:

    make deploy

### CODE REVIEW

**Comment by captbaritone** [comment]
Mabye @thejsj could sanity check this.

File: db/plugins.py [hunk 1]
```diff
@@ -555,12 +555,12 @@ def get_search_index():
...
-        r.row['slug'] != 'neobundle-vim',
-        r.row['slug'] != 'neobundle-vim-back-to-december',
-        r.row['slug'] != 'pathogen-vim',
-    ))
+    query = query.filter(
+        (r.row['slug'] != 'vundle') &
+        (r.row['slug'] != 'neobundle-vim') &
+        (r.row['slug'] != 'neobundle-vim-back-to-december') &
+        (r.row['slug'] != 'pathogen-vim')
+    )
```
Comment (thejsj) [inline db/plugins.py]: I'm not super familiar with the Python syntax, but you could rewrite this with `contains`. Something like: 

```
query = query.filter(
  lambda row: r.expr(['vundle', 'neobundle-vim', 'neobundle-vim-back-to-december', 'pathogen-vim']).contains(r.row['slug']).not()
)
```
Then you could move out that array out of the query:
```
unallowed_slugs = ['vundle', 'neobundle-vim', 'neobundle-vim-back-to-december', 'pathogen-vim']
query = query.filter(
  lambda row: r.expr(slugs).contains(r.row['slug']).not()
)
```

Comment (thejsj) [inline db/plugins.py]: Fixed query:
```
unallowed_slugs = ['vundle', 'neobundle-vim', 'neobundle-vim-back-to-december', 'pathogen-vim']
query = query.filter(
   lambda row: r.expr(unallowed_slugs).contains(row['slug']).not_())
)
```

**Comment by thejsj** [comment]
This seems reasonable and makes sense. Just FYI, Rethinkdb 2.3.0 was release about a week ago and contains some good changes like user accounts and built-in TLS support: http://rethinkdb.com/blog/2.3-release/

**Comment by captbaritone** [comment]
@thejsj Thanks! I looked at 2.3.0, but from what I can tell the Python library is not yet available from [pypy](https://pypi.python.org/pypi). Hopefully getting to 2.2.0 should allow us to jump to 2.3.0 with less risk.

**Comment by thejsj** [comment]
Oh, interesting. Yeah, that should definitely be the case. Still, 1.4 to
2.2 is a really nice upgrade!

Jorge Silva
blog.hiphipjorge.com
hiphipjorge.com
github.com/thejsj

**Comment by captbaritone** [comment]
@thejsj Updated. Thanks!

**Comment by thejsj** [comment]
This LGTM. I'd go ahead and merge it.

**Comment by captbaritone** [comment]
I'll run the deploy later today.

**Comment by dalanmiller** [comment]
If this hasn't been deployed yet, we just released RethinkDB 2.3.1, you should upgrade once more before deploying.

**Comment by thejsj** [comment]
@dalanmiller We can't upgrade to 2.3.1 because of pypy:

> @thejsj Thanks! I looked at 2.3.0, but from what I can tell the Python library is not yet available from pypy. Hopefully getting to 2.2.0 should allow us to jump to 2.3.0 with less risk.

https://github.com/vim-awesome/vim-awesome/pull/141#issuecomment-208428898

**Comment by dalanmiller** [comment]
You mean PyPI? 

This seems to be the most recent version. 

https://pypi.python.org/pypi/rethinkdb/2.3.0.post1

**Pull request merged**